From 7ceec1ddac79e6e7e3dc8b6c8f02fb01e2cbdc46 Mon Sep 17 00:00:00 2001
From: Vicente Adolfo Bolea Sanchez <vicente.bolea@kitware.com>
Date: Mon, 10 Oct 2022 13:47:46 -0400
Subject: [PATCH] libs: proper version and soversion in vtk-m libs

This commit reflects the changes of VTK-m API/ABI release plans in
which we attempt ABI compatibility in patch releases.

- It also sets the version/soname for diy thirdparty module lib
---
 CMake/VTKmWrappers.cmake           |  4 ++--
 vtkm/thirdparty/diy/CMakeLists.txt | 10 +++++++---
 2 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/CMake/VTKmWrappers.cmake b/CMake/VTKmWrappers.cmake
index 52bdede9b..3db410a58 100644
--- a/CMake/VTKmWrappers.cmake
+++ b/CMake/VTKmWrappers.cmake
@@ -499,8 +499,8 @@ function(vtkm_library)
 
   if (NOT VTKm_SKIP_LIBRARY_VERSIONS)
     # Setup the SOVERSION and VERSION information for this vtkm library
-    set_property(TARGET ${lib_name} PROPERTY VERSION 1)
-    set_property(TARGET ${lib_name} PROPERTY SOVERSION 1)
+    set_property(TARGET ${lib_name} PROPERTY VERSION ${VTKm_VERSION}.${VTKm_VERSION_PATCH})
+    set_property(TARGET ${lib_name} PROPERTY SOVERSION ${VTKm_VERSION})
   endif ()
 
   # Support custom library suffix names, for other projects wanting to inject
diff --git a/vtkm/thirdparty/diy/CMakeLists.txt b/vtkm/thirdparty/diy/CMakeLists.txt
index 2e273ecd9..ce29dfb34 100644
--- a/vtkm/thirdparty/diy/CMakeLists.txt
+++ b/vtkm/thirdparty/diy/CMakeLists.txt
@@ -58,7 +58,11 @@ set(VTKM_USE_EXTERNAL_DIY OFF)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Configure.h.in
   ${VTKm_BINARY_INCLUDE_DIR}/${kit_dir}/Configure.h)
 
-function(vtkm_diy_set_target_output_directory target)
+function(vtkm_diy_set_target_properties target)
+  if (NOT VTKm_SKIP_LIBRARY_VERSIONS)
+    set_property(TARGET ${target} PROPERTY VERSION ${VTKm_VERSION}.${VTKm_VERSION_PATCH})
+    set_property(TARGET ${target} PROPERTY SOVERSION ${VTKm_VERSION})
+  endif()
   set_property(TARGET ${target} PROPERTY ARCHIVE_OUTPUT_DIRECTORY ${VTKm_LIBRARY_OUTPUT_PATH})
   set_property(TARGET ${target} PROPERTY LIBRARY_OUTPUT_DIRECTORY ${VTKm_LIBRARY_OUTPUT_PATH})
   set_property(TARGET ${target} PROPERTY RUNTIME_OUTPUT_DIRECTORY ${VTKm_EXECUTABLE_OUTPUT_PATH})
@@ -69,10 +73,10 @@ add_subdirectory(vtkmdiy)
 
 # move diy libraries
 if (TARGET vtkmdiympi)
-  vtkm_diy_set_target_output_directory(vtkmdiympi)
+  vtkm_diy_set_target_properties(vtkmdiympi)
 endif()
 if (TARGET vtkmdiympi_nompi)
-  vtkm_diy_set_target_output_directory(vtkmdiympi_nompi)
+  vtkm_diy_set_target_properties(vtkmdiympi_nompi)
 endif()
 
 include(VTKmDIYUtils)
-- 
GitLab

